pgmpy/sampling/Sampling.py

```diff
@@ -67,19 +75,29 @@ def forward_sample(self, size=1):
         0        1           0          1
         1        1           0          2
         """
-        sampled = DataFrame(index=range(size), columns=self.topological_order)
+        types = [(var_name, 'int') for var_name in self.topological_order]
+        sampled = np.zeros(size, dtype=types).view(np.recarray)
+
         for node in self.topological_order:
             cpd = self.model.get_cpds(node)
             states = range(self.cardinality[node])
             evidence = cpd.variables[:0:-1]
             if evidence:
                 cached_values = self.pre_compute_reduce(variable=node)
-                evidence = sampled.ix[:, evidence].values
-                weights = list(map(lambda t: cached_values[tuple(t)], evidence))
+                evidence = np.vstack([sampled[i] for i in evidence])
+                weights = list(map(lambda t: cached_values[tuple(t)], evidence.T))
             else:
                 weights = cpd.values
             sampled[node] = sample_discrete(states, weights, size)
-        return sampled
+
+        if return_type.lower() == "dataframe":
```

Comment:
@khalibartan I am not too sure about this. But this block has been repeated in every method. Can we somehow devise a way to not make it redundant?